scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/definition/action/CommonModelDataInfoProvider.scala

```diff
@@ -26,18 +31,65 @@ abstract class CommonModelDataInfoProvider(modelData: ModelData) {
     modelData.modelClassLoader,
     Seq.empty,
     ProductionServiceInvocationCollector,
-    ComponentUseCase.TestDataGeneration,
+    RuntimeMode.Live,
+    NodesDeploymentData.empty,
     nonServicesLazyParamStrategy = LazyParameterCreationStrategy.default
   )
 
-  protected def compileSourceNode(
+  private lazy val subGraphCompiler = new PartSubGraphCompiler(nodeCompiler)
+
+  private lazy val scenarioCompiler = new ProcessCompiler(
+    modelData.modelClassLoader,
+    subGraphCompiler,
+    GlobalVariablesPreparer(modelData.modelDefinitionWithClasses.modelDefinition.expressionConfig),
+    nodeCompiler,
+    modelData.customProcessValidator
+  )
+
+  def collectAllSources(scenario: CanonicalProcess): List[SourceNodeData] = {
+    scenario.collectAllNodes.flatMap(asSource) ++ scenario.collectAllNodes.flatMap(asFragmentInputDefinition)
+  }
+
+  def compileSourceNode(
       source: SourceNodeData
   )(implicit jobData: JobData, nodeId: NodeId): ValidatedNel[ProcessCompilationError, Source] = {
     nodeCompiler.compileSource(source).compiledObject
   }
 
-  protected def collectAllSources(scenario: CanonicalProcess): List[SourceNodeData] = {
-    scenario.collectAllNodes.flatMap(asSource) ++ scenario.collectAllNodes.flatMap(asFragmentInputDefinition)
+  def compileScenario(scenario: CanonicalProcess)(implicit jobData: JobData): Map[NodeComponentInfo, Any] = {
```

Comment:
I think that it is important to emphasize somehow (maybe in method name?) that the purpose of this compilation is collection of all custom nodes (not filter or variable). Looking at current name it looks like we compile scenario to some executable unit. Maybe let's rename it to `compileAllCustomNodes`? Also this fallback to `Map.empty` is tricky. Maybe let's move it to place of invocation for clarity?